test/services/OdsServiceTest.scala

```diff
@@ -46,71 +49,69 @@ class OdsServiceTest extends UnitSpec with MockitoSugar with ScalaFutures {
 
     "return a successful future" in new TestService {
 
+      val fakeTaxSummaryLiability =
+        TaxSummaryLiability(
+          2014,
+          PensionTaxRate(5.0),
+          None,
+          Map(IncomeTaxDue -> Amount(BigDecimal(12.0), "gbp")),
+          Map(IncomeTaxDue -> Amount(BigDecimal(12.0), "gbp"))
+        )
+
       when(odsConnector.connectToSATaxpayerDetails(eqTo(testUtr))(any[HeaderCarrier]))
-        .thenReturn(Future.successful(mock[JsValue]))
+        .thenReturn(Future.successful(Some(mock[SaTaxpayerDetails])))
       when(odsConnector.connectToSelfAssessment(eqTo(testUtr), eqTo(2014))(any[HeaderCarrier]))
-        .thenReturn(Future.successful(mock[JsValue]))
-      when(jsonHelper.getAllATSData(any[JsValue], any[JsValue], eqTo(testUtr), eqTo(2014)))
+        .thenReturn(Future.successful(Some(fakeTaxSummaryLiability)))
+      when(jsonHelper.getAllATSData(any[SaTaxpayerDetails], any[TaxSummaryLiability], eqTo(testUtr), eqTo(2014)))
         .thenReturn(mock[JsValue])
 
       val result = getPayload(testUtr, 2014)(mock[HeaderCarrier])
 
       whenReady(result) { result =>
-        result shouldBe a[JsValue]
+        result.get shouldBe a[JsValue]
```

Comment:
Could this be a

`result.fold(fail("result is None"){ r => r should be a [JsValue] ...}`

Reply:
:+1: